virtio-devices/src/block.rs

```diff
@@ -232,13 +258,43 @@ impl BlockEpollHandler {
                 .ok_or(Error::MissingEntryRequestList)?;
             request.complete_async().map_err(Error::RequestCompleting)?;
 
+            let latency = request.start.elapsed().as_micros() as u64;
+            let read_ops_lst = self.counters.read_ops.load(Ordering::Relaxed) as i64;
+            let write_ops_lst = self.counters.write_ops.load(Ordering::Relaxed) as i64;
+            let mut read_avg =
+                self.counters.read_latency_avg.load(Ordering::Relaxed) as i64 * LATENCY_SCALE;
+            let mut write_avg =
+                self.counters.write_latency_avg.load(Ordering::Relaxed) as i64 * LATENCY_SCALE;
             let (status, len) = if result >= 0 {
                 match request.request_type {
                     RequestType::In => {
                         for (_, data_len) in &request.data_descriptors {
                             read_bytes += Wrapping(*data_len as u64);
                         }
                         read_ops += Wrapping(1);
+                        match latency {
```

Comment:
This is a strange way of doing it. Why use `match` like that? Just do two if statements instead, that will remove the need to extra check on if the max is uninitialised.

e.g.

```rust

if latency < self.counters.read_latency_min.load(Ordering::Relaxed) {
     self.counters.read_latency_min.store(latency, Ordering::Relaxed);
}

if latency > self.counters.read_latency_max.load(Ordering::Relaxed) {
     self.counters.read_latency_max.store(latency, Ordering::Relaxed);
}
```